support-frontend/app/utils/CheckoutValidationRules.scala

```diff
@@ -103,18 +109,23 @@ object CheckoutValidationRules {
         )
     }
 
-  def validate(createSupportWorkersRequest: CreateSupportWorkersRequest): Result =
-    (createSupportWorkersRequest.product match {
-      case _: SupporterPlus => PaidProductValidation.passes(createSupportWorkersRequest)
-      case d: DigitalPack => DigitalPackValidation.passes(createSupportWorkersRequest, d)
-      case p: Paper => PaperValidation.passes(createSupportWorkersRequest, p.fulfilmentOptions)
-      case _: GuardianWeekly => GuardianWeeklyValidation.passes(createSupportWorkersRequest)
-      case _: Contribution => PaidProductValidation.passes(createSupportWorkersRequest)
-    }) match {
-      case Invalid(message) =>
-        Invalid(s"validation of the request body failed with $message - body was $createSupportWorkersRequest")
-      case Valid => Valid
+  def validate(
+      createSupportWorkersRequest: CreateSupportWorkersRequest,
+      paperRound: PaperRoundAPI,
+  )(implicit ec: ExecutionContext): Future[Result] = for {
+    result <- createSupportWorkersRequest.product match {
+      case _: SupporterPlus => Future { PaidProductValidation.passes(createSupportWorkersRequest) }
```

Comment:
this change is just to stop it spinning off the contained code in a separate thread
```suggestion
      case _: SupporterPlus => Future.successful { PaidProductValidation.passes(createSupportWorkersRequest) }
```

Reply:
Ah perfect, I wondered about that alright!

Reply:
Done :)